addons/mail/models/discuss/ir_binary.py

```diff
@@ -0,0 +1,22 @@
+from odoo import models
+
+
+class IrBinary(models.AbstractModel):
+    _inherit = "ir.binary"
+
+    def _find_record_check_access(self, record, access_token, field):
+        if record._name in ["res.partner", "mail.guest"] and field == "avatar_128":
+            current_partner, current_guest = self.env["res.partner"]._get_current_persona()
+            record_rule = []
+            if record._name == "res.partner":
+                record_rule.append(("channel_member_ids", "any", [("partner_id", "=", record.id)]))
+            else:
+                record_rule.append(("channel_member_ids", "any", [("guest_id", "=", record.id)]))
+            if current_partner:
+                record_rule.append(("channel_member_ids", "any", [("partner_id", "=", current_partner.id)]))
+            else:
+                record_rule.append(("channel_member_ids", "any", [("guest_id", "=", current_guest.id)]))
+            allow_access = self.env["discuss.channel"].search_count(record_rule)
```

Comment:
FYI, you can add a limit to `search_count` when you only want to check the existence of a record. Generally speaking, it improves the performance of the query (but it is not always true).
```suggestion
            allow_access = self.env["discuss.channel"].search_count(record_rule, limit=1)
```